### ISSUE TITLE
RESUME FROM in the config doesnt work

### ISSUE THREAD

**Issue by YuktiADY** [open-mmlab/mmpose#1351]
Hallo ,

I already trained hrnet[w32]_256x192 model and now i want to again continue training for some epochs so i gave in the config.
resume_from = '/home/yukti/mmpose/theodore_2022-04-25/best_AP_epoch_30.pth'

but the training is not resuming from this epoch rather ti shows this .

2022-05-03 14:47:57,328 - mmpose - INFO - workflow: [('train', 1)], max: 30 epochs
2022-05-03 14:47:57,328 - mmpose - INFO - Checkpoints will be saved to /home/yukti/mmpose/theodore_2022-05-03 by HardDiskBackend.
INFO:torch.distributed.elastic.agent.server.api:[default] worker group successfully finished. Waiting 300 seconds for other agents to finish.
INFO:torch.distributed.elastic.agent.server.api:Local worker group finished (SUCCEEDED). Waiting 300 seconds for other agents to finish
/home/yukti/anaconda3/envs/open-mmlab/lib/python3.7/site-packages/torch/distributed/elastic/utils/store.py:71: FutureWarning: This is an experimental API and will be changed in future.
  "This is an experimental API and will be changed in future.", FutureWarning
INFO:torch.distributed.elastic.agent.server.api:Done waiting for other agents. Elapsed: 0.0006282329559326172 seconds
{"name": "torchelastic.worker.status.SUCCEEDED", "source": "WORKER", "timestamp": 0, "metadata": {"run_id": "none", "global_rank": 0, "group_rank": 0, "worker_id": "3529801", "role": "default", "hostname": "dst-toaster.etit.tu-chemnitz.de", "state": "SUCCEEDED", "total_run_time": 45, "rdzv_backend": "static", "raw_error": null, "metadata": "{\"group_world_size\": 1, \"entry_point\": \"python\", \"local_rank\": [0], \"role_rank\": [0], \"role_world_size\": [1]}", "agent_restarts": 0}}
{"name": "torchelastic.worker.status.SUCCEEDED", "source": "AGENT", "timestamp": 0, "metadata": {"run_id": "none", "global_rank": null, "group_rank": 0, "worker_id": null, "role": "default", "hostname": "dst-toaster.etit.tu-chemnitz.de", "state": "SUCCEEDED", "total_run_time": 45, "rdzv_backend": "static", "raw_error": null, "metadata": "{\"group_world_size\": 1, \"entry_point\": \"python\"}", "agent_restarts": 0}}

Last time ti worked but not this time. Please suggest!

**Comment by liqikai9** [comment]
@YuktiADY Thanks for your feedback. It seems like a warning and should not affect. 

I think it might be related to the **compatibility** between `CUDA`, `PyTorch `and `mmcv-full`. Please double their versions.

**Comment by YuktiADY** [comment]
Thanks for the reply. The same config when given resume_from worked but now i changed in the config the learning scheduler.
How can i double the versions ?

**Comment by liqikai9** [comment]
Sorry, there is a typo. I was saying that you can **double-check** the versions of  `CUDA`, `PyTorch` and `mmcv-full`.

**Comment by liqikai9** [comment]
> The same config when given resume_from worked but now i changed in the config the learning scheduler.

If you want to use the **same** learning scheduler and the checkpoint is trained with the same config, you don't need to adjust the learning scheduler manually, the code will **automatically** load the checkpoint and adjust the learning rate according to the learning scheduler in the config and trained epochs. 

You can check the source code in mmcv for more detail: https://github.com/open-mmlab/mmcv/blob/a3a078eefb703cc22412af0d133e9c378cb3f56f/mmcv/runner/base_runner.py#L347

**Comment by YuktiADY** [comment]
> 

I have checked the versions CUDA = 11.4, Pytorch = 1.9.0.. mmcv-full version how to check ?

**Comment by YuktiADY** [comment]
> > The same config when given resume_from worked but now i changed in the config the learning scheduler.
> 
> If you want to use the **same** learning scheduler and the checkpoint is trained with the same config, you don't need to adjust the learning scheduler manually, the code will **automatically** load the checkpoint and adjust the learning rate according to the learning scheduler in the config and trained epochs.
> 
> You can check the source code in mmcv for more detail: https://github.com/open-mmlab/mmcv/blob/a3a078eefb703cc22412af0d133e9c378cb3f56f/mmcv/runner/base_runner.py#L347

I meant to say earlier when using the same config resume from was working , now this i trained the same config but only changed learning scheduler , so does this scheduler has any impact on this  resume from ?

**Comment by liqikai9** [comment]
You can run `PYTHONPATH=${PWD}:$PYTHONPATH python mmpose/utils/collect_env.py` to collect necessary environment information, including the mmcv version.

> so does this scheduler has any impact on this resume from ?

Yes, it will affect. How did you modify the config?

**Comment by YuktiADY** [comment]
> You can run `PYTHONPATH=${PWD}:$PYTHONPATH python mmpose/utils/collect_env.py` to collect necessary environment information, including the mmcv version.
>
> > so does this scheduler has any impact on this resume from ?
> 
> Yes, it will affect. How did you modify the config?

I manually made changes in the config.
Is the path ok. I gave but its says no such file or dicrectory.

**Comment by liqikai9** [comment]
The config looks fine. Originally, I meant to say earlier that what was the command you used to run the training task? Could you please offer that too? Thanks. 

BTW, what is the output of the following command?

> yukti@dst-toaster:~/mmpose$ python mmpose/utils/collect_env.py

**Comment by YuktiADY** [comment]
I also one question .If there is a possibility to augment the data with random noise in the code of mmpose and where we can add?
How to check this ?

**Comment by YuktiADY** [comment]
> I also have  one more question .If there is a possibility to augment the data with random noise in the code of mmpose and where we can add? How to check the possibility  ?

**Comment by liqikai9** [comment]
Hi, can you resume the training using the trained checkpoint and the same config now? Is there anything I can help with?

**Comment by YuktiADY** [comment]
The training is already going on , should i stop and resume ? Do you think it will work ? Because we didnt do any changes .

**Comment by liqikai9** [comment]
Did you set the resume field as the checkpoint path in the config? Did the `resume` work now?

**Comment by liqikai9** [comment]
Seems like you train it without resuming from the previous checkpoint and the training goes well. If the training stop unexpectedly, you can put the output log here so that we can localize the possible problem.

**Comment by YuktiADY** [comment]
I have just cancelled my training which was going on and set the resume field from checkpoint path which was done few days back.
I found something new .
Case 1. 
resume_from = '/home/yukti/mmpose/theodore_2022-04-25/epoch_30.pth'
total_epochs = 60
Its is resumed and starts trainig  from epoch[31]

Case 2 .
resume_from = '/home/yukti/mmpose/theodore_2022-04-25/epoch_30.pth'
total_epochs = 30

Gets that warning and training is completed very fast.

**Comment by liqikai9** [comment]
Yep. That goes well. If you use `resume_from`, the training will be continued from the latest epoch, which will be recorded in this checkpoint:  '/home/yukti/mmpose/theodore_2022-04-25/epoch_30.pth'. 
So if the total epochs is 60, the training will start from epoch 31, and the learning rate will also be tuned. 

If you just want to load the checkpoint and start the training from epoch 1, you should set: 
`load_from = '/home/yukti/mmpose/theodore_2022-04-25/epoch_30.pth'` instead of `resume_from`.

Be careful to these two fields: `load_from`  and `resume_from`, see here: https://github.com/open-mmlab/mmpose/blob/master/docs/en/tutorials/0_config.md#config-system
